torch/_dynamo/debug_utils.py

```diff
@@ -698,3 +701,76 @@ def symint(self, name, val) -> None:
         if isinstance(val, torch.SymInt):
             val = val.node.hint
         self._lines.append(f"reader.symint({val!r})  # {name}")
+
+
+def aot_graph_input_parser(
+    func: Callable[[List[Tensor]], List[Tensor]], device="cuda"
+) -> Dict[str, Any]:
+    """
+    Takes in a function which has been printed with print_readable() and constructs kwargs to run it.
+
+    Currently only handles Tensor inputs and a graph module which might have tensor constants.
+
+
+    Example:
+        Consider a function `forward` defined as follows:
+
+        >>> def forward(self, primals_1: "f32[1001, 6]"):
+        ...     _tensor_constant0: "i64[4190]" = self._tensor_constant0
+        ...     # Further implementation
+
+        >>> kwargs = aot_graph_input_parser(forward)
+        >>> forward(**kwargs)
+    """
+
+    from torch.fx.graph import dtype_abbrs
+
+    dtype_map = {value: key for key, value in dtype_abbrs.items()}
+    dtype_pattern = "|".join(dtype_abbrs.values())
+
+    # Extracting the source code from the function
+    source = inspect.getsource(func)
+
+    # Regular expressions
+    tensor_assignment_regex = rf"(_tensor_constant\d+): \"({dtype_pattern})\[\s*(.*?)\s*\]\" = self\.(_tensor_constant\d+)"
+
+    tensor_regex = rf"({dtype_pattern})\[\s*(.*?)\s*\]"
+
+    class TensorContainer:
+        "Container for tensors as attributes"
+        pass
+
+    container = TensorContainer()
+    # Dictionary for tensors from annotations
+    kwargs = {}
+
+    def gen_tensor(shape, dtype):
+        shape = tuple(map(int, shape_str.split(",")))
```

Comment:
This line can be removed since the two callers already does the string parsing work